drape_frontend/gui/speed_limit.cpp

```diff
@@ -0,0 +1,102 @@
+#include "drape_frontend/gui/speed_limit.hpp"
+
+#include "drape_frontend/animation/show_hide_animation.hpp"
+#include "drape_frontend/gui/drape_gui.hpp"
+
+#include "shaders/programs.hpp"
+
+#include "drape/glsl_func.hpp"
+#include "drape/glsl_types.hpp"
+
+#include <functional>
+#include <utility>
+
+using namespace std::placeholders;
+
+namespace gui
+{
+namespace
+{
+class TextHandle : public MutableLabelHandle
+{
+  using TBase = MutableLabelHandle;
+
+public:
+  TextHandle(uint32_t id, m2::PointF const & pivot, ref_ptr<dp::TextureManager> textures)
+    : TBase(id, dp::Center, pivot)
+  {
+    SetTextureManager(textures);
+  }
+
+private:
+  bool Update(ScreenBase const & screen) override
+  {
+    SpeedLimitHelper const & helper = DrapeGui::Instance().GetSpeedLimitHelper();
+    SetIsVisible(helper.IsSpeedLimitAvailable());
+    if (IsVisible())
+    {
+      SetContent(helper.GetSpeedLimit());
+    }
+
+    return TBase::Update(screen);
+  }
+};
+
+class BackgroundHandle : public elements::CircleHandle
+{
+public:
+  using CircleHandle::CircleHandle;
+
+  bool Update(const ScreenBase & screen) override
+  {
+    SpeedLimitHelper const & helper = DrapeGui::Instance().GetSpeedLimitHelper();
+    SetIsVisible(helper.IsSpeedLimitAvailable());
+    return CircleHandle::Update(screen);
+  }
+};
+}  // namespace
+
+SpeedLimit::SpeedLimit(const Position & position) : Shape(position)
+{
+  m_background.SetHandleId(EGuiHandle::GuiHandleSpeedSign);
+  m_background.SetPosition(position);
+  m_background.SetRadius(Config::kBackgroundRadius);
+  m_background.SetOutlineWidthRatio(Config::kBackgroundOutlineWidthRatio);
+  m_background.SetColor(Config::kBackgroundColor);
+  m_background.SetOutlineColor(Config::kBackgroundOutlineColor);
+  m_background.SetHandleCreator(
+      [](uint32_t id, dp::Anchor anchor, m2::PointF const & pivot, m2::PointF const & size)
+      { return make_unique_dp<BackgroundHandle>(id, anchor, pivot, size); });
```

Comment:
Hm, probably not related to this PR (or GUI elements arch), but from my perspective, the whole SpeedLimit should be one GUI-handle element, but we create separate handles, (text, circle) ..

Reply:
Do you have any idea or example how to make only one handle for many shapes?